spec/draper/query_methods_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'spec_helper'
+
+module Draper
+  describe QueryMethods do
+    describe "#method_missing" do
+      let(:instance) { Struct.new(:query_methods){include Draper::QueryMethods}.new(double) }
+      let(:fake_strategy) { instance_double(QueryMethods::LoadStrategy) }
+
+      before { allow(QueryMethods::LoadStrategy).to receive(:new).and_return(fake_strategy) }
+
+      context 'When strategy allows instance to call the method' do
+        before { allow(fake_strategy).to receive(:allowed?).with(:foo).and_return(true) }
+
+        it 'calls the method' do
+          allow(instance).to receive(:foo)
```

Comment:
this can't be on the before?

Reply:
I prefer to write the specs using the format: setup/exercise/verify.